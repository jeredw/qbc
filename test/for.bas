for i=1 to 20
print "hi"
next i
for i=10 to 1 step -1
next
for a=1 to 5
for b=1 to 5
next b
next a
for a=1 to 5
for b=1 to 5
next b, a
print "hi": for i=1 to 2: print "hi"
print "foo": next
for i=1 to 10: for j=1 to 10: print i, j: next j: next i
for i=1 to 42: next: print "statement"
for i=1 to 42: next ' i am a comment
