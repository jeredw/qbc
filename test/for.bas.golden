(program 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 20)) 
      (block \n 
        (statement 
          (print_statement print 
            (expr 
              (literal "hi")))) \n statement) next i)) \n 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 10)) to 
      (expr 
        (literal 1)) step 
      (expr - 
        (expr 
          (literal 1))) 
      (block \n statement) next)) \n 
  (statement 
    (for_next_statement for a = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 5)) 
      (block \n 
        (statement 
          (for_next_statement for b = 
            (expr 
              (literal 1)) to 
            (expr 
              (literal 5)) 
            (block \n statement) next b)) \n statement) next a)) \n 
  (statement 
    (for_next_statement for a = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 5)) 
      (block \n 
        (statement 
          (for_next_statement for b = 
            (expr 
              (literal 1)) to 
            (expr 
              (literal 5)) 
            (block \n statement) next b))) , a)) \n 
  (statement 
    (print_statement print 
      (expr 
        (literal "hi")))) : 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 2)) 
      (block : 
        (statement 
          (print_statement print 
            (expr 
              (literal "hi")))) \n 
        (statement 
          (print_statement print 
            (expr 
              (literal "foo")))) : statement) next)) \n 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 10)) 
      (block : 
        (statement 
          (for_next_statement for j = 
            (expr 
              (literal 1)) to 
            (expr 
              (literal 10)) 
            (block : 
              (statement 
                (print_statement print 
                  (expr 
                    (variable_or_function_call i)) , 
                  (expr 
                    (variable_or_function_call j)))) :) next j)) :) next i)) \n 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 42)) 
      (block :) next)) : 
  (statement 
    (print_statement print 
      (expr 
        (literal "statement")))) \n 
  (statement 
    (for_next_statement for i = 
      (expr 
        (literal 1)) to 
      (expr 
        (literal 42)) 
      (block :) next)) \n)
